documentation/concepts.md

```diff
@@ -18,28 +18,57 @@ is then exposed to subscribers. Specifically, Liftbridge centers around the
 concept of a *stream*, which is a durable message stream attached to a NATS
 subject. A stream consists of one or more *partitions*, which are ordered,
 replicated, and durably stored on disk and serve as the unit of storage and
-parallelism in Liftbridge.
+parallelism in Liftbridge. [@tyler Pls elaborate on where the log is stored exactly.
+This is important since users will probably want attach a k8s storage to the volume->dir, eg. s3 bucket or SSDs. This will enable statefulness when using stateful sets in k8s.]
+
+[@Tyler The dependency on NATS also means that future changes to the NATS project will impact the Liftbridge project. This problem will be remediated by versioning of stable releases of Liftbridge pointing to a stable release of NATS?]
 
 Streams have a few key properties: a subject, which is the corresponding NATS
 subject, a name, which is a human-readable identifier for the stream, and a
 replication factor, which is the number of nodes the stream's partitions should
 be replicated to for redundancy. Optionally, there is a group which is the name
 of a load-balance group for the stream to join. When there are multiple streams
-in the same group, messages will be balanced among them.
+in the same group, messages will be balanced among them. 
+
+```
+Use case note:
+The usual use case is as follows. A typical subject is the command subject (compare kafka topic) of the CQRS pattern. The corresponding log created by this
+subject is the implementation of the event sourcing pattern. The response of a command being put on a subject is a microservice worker reading that command 
+off the subject and executing the command. Subsequently, the result of this activity is then posted on another subject, perhaps for down streams analytical 
+reporting purposes etc. This enables the Query in the CQRS patterns. A careful reader saw the above remark of replication factor, redundancy and groups. 
+These choices will impact this example with microservice workers, since the order of messages and guarantees on replication will be affected by these chioces. More on that later.
+
+For further info on CQRS and event sourceing pls
+see https://martinfowler.com/bliki/CQRS.html
+and https://martinfowler.com/eaaDev/EventSourcing.html
+respectively. 
+```
 
 There can be multiple streams attached to the same NATS subject, but stream
-names must be unique within a cluster.
+names must be unique within a cluster. 
+
+[@Tyler Please describe if this is intended architecture or just happened due to dependecy towards NATS. If the latter we need to advice on usage. If the former, we need to descibe the use case. ]
+```
+This means that the concept of stream name is now defined as being the least common denominator, not the NATS subject as when using NATS alone without Liftbridge.
+```
```

Comment:
I might have misunderstood, but how I interpret the text is as follows. Since Liftbridge can have streams aggregate 1..N NATS subjects and 1 NATS subject can contain 1..N Liftbridge streams we need advise on why one would choose a particular setup. This means one set up/configuration per use case. Also, one should point out choices of configuration that one should not use. Imagine the Liftbridge user reading the text. Identifies his/her use case and get an advise on architectural setup. That is real value. Do you agree?